### ISSUE TITLE
question: is the original original authority before host rewrite?

### ISSUE THREAD

**Issue by marcosrmendezthd** [envoyproxy/envoy#13834]
**If you are reporting *any* crash or *any* potential security issue, *do not*
open an issue in this repo. Please report the issue via emailing
envoy-security@googlegroups.com where the issue will be triaged appropriately.**

*Title*: *is the original original authority before host rewrite?*

*Description*:
I'm trying to add the original destination (authority) of a request via an access log, or even an additional request header before it gets rewritten for the upstream cluster request. Is this information available somewhere?

An access log with... still logs the rewritten host.

action.Route.HostRewriteSpecifier = &envoyroute.RouteAction_HostRewrite{
	HostRewrite: "my new upstream host,
}

```
"req_host":     "%REQ(:AUTHORITY)%"
```

We can use "%REQ(X-ENVOY-ORIGINAL-PATH?:PATH)%" for the original path? Is there something like that for the original authority?

**Comment by marcosrmendezthd** [comment]
thanks @mattklein123 !

**Comment by marcosrmendezthd** [comment]
any updates on this?

**Comment by mattklein123** [comment]
I thought someone might have added this but I don't see it. Seems like a reasonable request to me. cc @alyssawilk

**Comment by alyssawilk** [comment]
Yeah, I agree this is reasonable, but I've got a few other things in my queue so I'll leave it as help wanted for now.

**Comment by 1it** [comment]
I have the same issue, it would be very much appreciated to see that feature.

**Comment by marcosrmendezthd** [comment]
any updates on this @alyssawilk ?

**Comment by alyssawilk** [comment]
no updates from me as I'm not working on it.  I believe no one has picked it up still.
